src/hardware/gwinstek-gds-800/api.c

```diff
@@ -71,24 +121,58 @@ static struct sr_dev_inst *probe_device(struct sr_scpi_dev_inst *scpi)
 	devc->frame_limit = 1;
 	devc->sample_rate = 0.0;
 	devc->df_started = FALSE;
+	if ( strncmp(sdi->model, "GDS-2", 5) == 0 &&
+		strlen(sdi->model) == 8 &&
+		sdi->model[7] == '4' ) {
```

Comment:
Please remove space before strncmp and wrap each expression in brackets to avoid possible operator precedence issues: if ((...) && (...) && (...)) {